test/unittests/execute_floating_point_test.cpp

```diff
@@ -439,6 +484,56 @@ TYPED_TEST(execute_floating_point_types, neg)
     }
 }
 
+TYPED_TEST(execute_floating_point_types, ceil)
+{
+    auto instance = instantiate(parse(this->get_unop_code(Instr::f32_ceil)));
+    const auto exec = [&](auto arg) { return execute(*instance, 0, {arg}); };
+
```

Comment:
I think there should be a specific case for `-q = -0 (if -1 < -q < 0)` here.

The rules about infinity and zero matches that of C++ too (https://en.cppreference.com/w/cpp/numeric/math/ceil), but stil having tests here would be needed.

Not sure the rule requires quiet_nan, but needs a test too, because C++ just passes through any nan.

Reply:
1. I can the special case.
2. Zeros and infinities are included in the `positive_trunc_tests`. Rules are simple and the same for all 4 rounding instructions, so I don't see point of having them duplicated everywhere.
3. NaNs are checked in `unop_nan_propagation` as this is common rules for all instructions (except some which ignore NaNs).

Reply:
Those special cases also exists for floor/trunc (actually 2 for trunc).

Will check `positive_trunc_tests`.

Reply:
I added this comment in the `positive_trunc_tests`.